### PR TITLE
Feat/autorenew payment mean

### PR DESCRIPTION
**Pull request by marie-j**
### Description of the Change

Display message to redirect to payment mean add form if users try to activate autorenew  and there is no valid payment mean 

MANAGER-1815
MANAGER-1818

### Related : 
https://github.com/ovh-ux/ovh-angular-sidebar-menu/pull/28

### CODE REVIEW

File: client/app/account/billing/autoRenew/warnPaymentMean/billing-autoRenew-warnPaymentMean.html [hunk 1]
```diff
@@ -0,0 +1,10 @@
+<div data-ng-controller="Billing.controllers.AutoRenew.WarnPaymentMean as $ctrl">
+    <oui-modal data-heading="{{ ::'autorenew_service_active_title' | translate }}"
+               data-primary-action="$ctrl.goToAddPaymentMean()"
+               data-primary-label="{{:: 'payment_mean_add_title' | translate }}"
+               data-secondary-action="$ctrl.close()"
+               data-secondary-label="{{ 'common_cancel' | translate }}"
+               on-dismiss="$ctrl.close()">
```
Comment (antleblanc) [inline client/app/account/billing/autoRenew/warnPaymentMean/billing-autoRenew-warnPaymentMean.html]:
```suggestion
               data-on-dismiss="$ctrl.close()">
```

File: client/app/account/billing/autoRenew/billing-autoRenew.controller.js [hunk 2]
```diff
@@ -725,58 +740,37 @@ angular.module('Billing.controllers').controller('Billing.controllers.AutoRenew'
...
+          user,
+          hasDefaultValidPaymentMean,
+          renewAlignUrl,
+          userGuide,
+          serviceTypes,
+          userCertificates,
+        }) => {
+          $scope.user = user;
+          $scope.urls.renewAlign = renewAlignUrl;
+          $scope.canDisableAllDomains = userCertificates.includes('domains-batch-autorenew');
```
Comment (antleblanc) [inline client/app/account/billing/autoRenew/billing-autoRenew.controller.js]: Does it make sense to turn this into an `angular.constant`? (`'domains-batch-autorenew'`)

File: client/app/account/billing/autoRenew/billing-autoRenew.controller.js [hunk 3]
```diff
@@ -725,58 +740,37 @@ angular.module('Billing.controllers').controller('Billing.controllers.AutoRenew'
...
+          $scope.hasDefaultValidPaymentMean = hasDefaultValidPaymentMean;
+          $scope.automaticRenewV2Mean.allowed = hasDefaultValidPaymentMean;
+
+          if (userGuide && userGuide.autoRenew) {
```
Comment (FredericEspiau) [inline client/app/account/billing/autoRenew/billing-autoRenew.controller.js]:
```suggestion
          if (_.has(userGuide, 'autoRenew') {
```

File: client/app/account/billing/autoRenew/warnPaymentMean/billing-autoRenew-warnPaymentMean.controller.js [hunk 4]
```diff
@@ -0,0 +1,14 @@
+angular.module('Billing.controllers').controller('Billing.controllers.AutoRenew.WarnPaymentMean', class {
```
Comment (FredericEspiau) [inline client/app/account/billing/autoRenew/warnPaymentMean/billing-autoRenew-warnPaymentMean.controller.js]:
```suggestion
angular.module('Billing.controllers').controller('billingControllersAutoRenewWarnPaymentMean', class {
```

File: client/app/account/billing/autoRenew/warnPaymentMean/billing-autoRenew-warnPaymentMean.html [hunk 5]
```diff
@@ -0,0 +1,10 @@
+<div data-ng-controller="Billing.controllers.AutoRenew.WarnPaymentMean as $ctrl">
+    <oui-modal data-heading="{{ ::'autorenew_service_active_title' | translate }}"
+               data-primary-action="$ctrl.goToAddPaymentMean()"
+               data-primary-label="{{:: 'payment_mean_add_title' | translate }}"
+               data-secondary-action="$ctrl.close()"
+               data-secondary-label="{{ 'common_cancel' | translate }}"
```
Comment (FredericEspiau) [inline client/app/account/billing/autoRenew/warnPaymentMean/billing-autoRenew-warnPaymentMean.html]:
```suggestion
               data-secondary-label="{{:: 'common_cancel' | translate }}"
```

File: client/app/account/billing/autoRenew/warnPaymentMean/billing-autoRenew-warnPaymentMean.html [hunk 6]
```diff
@@ -0,0 +1,10 @@
+<div data-ng-controller="Billing.controllers.AutoRenew.WarnPaymentMean as $ctrl">
+    <oui-modal data-heading="{{ ::'autorenew_service_active_title' | translate }}"
```
Comment (FredericEspiau) [inline client/app/account/billing/autoRenew/warnPaymentMean/billing-autoRenew-warnPaymentMean.html]:
```suggestion
    <oui-modal data-heading="{{:: 'autorenew_service_active_title' | translate }}"
```

Choose a convention and stick to it :)

**Pull request merged**